### PR TITLE
fix(es/codegen): Fix codegen of string literals with `\x000`

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue:**

 - Closes https://github.com/swc-project/swc/issues/6836.

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_codegen

File: crates/swc_ecma_codegen/src/lib.rs [hunk 1]
```diff
@@ -3481,7 +3481,7 @@ fn get_quoted_utf16(v: &str, ascii_only: bool, target: EsVersion) -> String {
 
                 match next {
                     Some('1'..='9') => buf.push_str("\\x00"),
-                    _ => buf.push_str("\\0"),
+                    _ => buf.push_str("\\x00"),
```
Comment (jridgewell) [inline crates/swc_ecma_codegen/src/lib.rs:3484]: I think the only problem was the range here:
```suggestion
                    Some('0'..='7') => buf.push_str("\\x00"),
                    _ => buf.push_str("\\0"),
```

Comment (kdy1) [reply crates/swc_ecma_codegen/src/lib.rs:3484]: Thank you!

**Review by jridgewell — APPROVED** [review]

File: crates/swc_ecma_codegen/src/lib.rs [hunk 2]
```diff
@@ -3480,7 +3480,7 @@ fn get_quoted_utf16(v: &str, ascii_only: bool, target: EsVersion) -> String {
                 let next = iter.peek();
 
                 match next {
-                    Some('1'..='9') => buf.push_str("\\x00"),
+                    Some('0'..='7') => buf.push_str("\\x00"),
```
Comment (jridgewell) [inline crates/swc_ecma_codegen/src/lib.rs:3483]: Sorry, no, it still fails!
```js
() => { "use strict"; 1; "\09"; }
```

**Review by jridgewell — CHANGES_REQUESTED** [review]

File: crates/swc_ecma_codegen/src/lib.rs (hunk 2 above)
Comment (jridgewell) [reply crates/swc_ecma_codegen/src/lib.rs:3483]: Yup: https://tc39.es/ecma262/multipage/ecmascript-language-lexical-grammar.html#prod-EscapeSequence

Sorry, it's gotta be `0..=9` to properly pass. `"\09"` is a [LegacyOctalEscapeSequence](https://tc39.es/ecma262/multipage/ecmascript-language-lexical-grammar.html#prod-LegacyOctalEscapeSequence), which is only valid in loose mode.

**Review by jridgewell — DISMISSED** [review]

**Pull request merged**